test/rules/ban-parens/test.tsx.lint

```diff
@@ -0,0 +1,9 @@
+const foo = (
+            ~
+    <div>text</div>
+~~~~~~~~~~~~~~~~~~~
+);
```

Comment:
these should not be banned by default. parens around multiline jsx statements are _enforced_ in the default tslint-react config. 

this is why I wanted you to add these tests -- the default config should not conflict with tslint-react.

Reply:
I see -- I wish you'd just said that.

I don't know if it'll be easy to do this, if jsx expressions can occur anywhere. That eliminates most of the rules. I can just exclude jsx files entirely from this check, maybe?

Reply:
Sorry if I wasn't clear upfront. Excluding `.tsx` files completely in this rule is not a good option because it will feel pretty inconsistent for TSX users -- there's lots of regular TS code in `.tsx` files. I haven't really dug into the internals of this rule implementation here (deferring to @ajafff) but I am concerned about the overall UX of this rule for tslint-react users, so I'd like you to update it before merging.